File: golf/smol.py
import json,sys;from subprocess import getoutput;from urllib.request import Request,urlopen;from uuid import uuid4
url=sys.argv[1];h=[];H={"Content-Type":"application/json","session_id":uuid4().hex};b=dict(model="gpt-5.6-sol",input=h,tools=[dict(type="custom",name="sh")])
while True:
  if not(p:=input("> ")).strip():continue
  h+=[dict(role="user",content=p)]
  while True:
    r=json.load(urlopen(Request(url,json.dumps(b).encode(),H)));o=r["output"];h+=o;c=[i for i in o if i["type"]=="custom_tool_call"]
    if not c:print(o[-1]["content"][0]["text"],f'\n[{r["usage"]["total_tokens"]/10500:05.2f}%]');break
    h+=[dict(type="custom_tool_call_output",call_id=i["call_id"],output=getoutput(i["input"])) for i in c]

File: golf/smol.clj
(require '[babashka.http-client :as h]'[babashka.process :as p]'[cheshire.core :as j])
(let [u(first *command-line-args*) k(str(random-uuid))]
 (loop [x []](print "> ")(flush)
  (when-let [s(read-line)](recur(if(.isBlank s)x
    (loop [x(conj x {:role "user" :content s})]
     (let [r(->(h/post u {:headers {:content-type "application/json" :session_id k}
                          :body(j/encode {:model "gpt-5.6-sol" :input x :tools [{:type "custom" :name "sh"}]})})
                 :body(j/decode true))
           o(:output r) x(into x o) c(filter #(="custom_tool_call"(:type %))o)]
      (if(seq c)(recur(into x(for [i c]
        (let [z(p/sh {:err :out} "/bin/sh" "-c"(:input i))]
         {:type "custom_tool_call_output" :call_id(:call_id i) :output(format "exit %d\n%s"(:exit z)(:out z))}))))
       (do(printf "%s\n[%05.2f%%]\n"(-> o last :content first :text)(/(-> r :usage :total_tokens)10500.0))x)))))))))
